core-bundle/tests/Twig/FailTolerantFilesystemLoaderTest.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Twig;
+
+use Contao\CoreBundle\Cache\ApplicationCacheState;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\CoreBundle\Twig\FailTolerantFilesystemLoader;
+use Twig\Loader\FilesystemLoader;
+use Webmozart\PathUtil\Path;
+
+class FailTolerantFilesystemLoaderTest extends TestCase
+{
+    /**
+     * @dataProvider providePaths
+     */
+    public function testAddPathDelegatesCallsIfCacheIsNotDirty(string $path): void
+    {
+        $inner = $this->createMock(FilesystemLoader::class);
+        $inner
+            ->expects($this->once())
+            ->method('addPath')
+            ->with($path, 'namespace')
+        ;
+
+        $cacheState = $this->createMock(ApplicationCacheState::class);
+        $cacheState
+            ->method('isDirty')
+            ->willReturn(false)
+        ;
+
+        $filesystemLoader = new FailTolerantFilesystemLoader(
+            $inner,
+            $cacheState,
+            $this->getTestRoot(),
+        );
+
+        $filesystemLoader->addPath($path, 'namespace');
+    }
+
+    /**
+     * todo: remove.
+     */
+    public function testCI(): void
+    {
+        // works
+        $this->assertDirectoryExists(Path::join($this->getTestRoot(), 'app'));
+
+        // fails
+        $this->assertDirectoryExists(Path::join($this->getTestRoot(), 'templates'));
```

Comment:
I have no clue why the tests are failing on CI? The template fixture folder I added seems missing. I created this temporary test to showcase the problem.

There are `.gitignore` files in them and they're clearly there if i clone the repo and checkout this branch....  What am I missing? :confused:

Reply:
The issue was indeed `core-bundle/tests/Contao/TemplateLoaderTest.php` removing the folder. :facepalm: 
Adjusted in 3b93f875f50bdfdacc51b1dd5e885e9c6e53c4da.